lib/generators/deimos/active_record_consumer_generator.rb

```diff
@@ -0,0 +1,151 @@
+
+      no_commands do
+
+        # Creates database migration for creating new table
+        def create_db_migration
+          migration_template('migration.rb', "#{db_migrate_path}/create_#{table_name.underscore}.rb")
+        end
+
+        # Creates Rails Model
+        def create_rails_model
+          template('model.rb', "app/models/#{model_name}.rb")
+        end
+
+        # Creates Kafka Consumer file
+        def create_consumer
+          template('consumer.rb', "app/lib/kafka/models/#{consumer_name.underscore}.rb")
+        end
+
+        # Adds consumer config to config file.
+        # Defaults to deimos.rb if config_path arg is not specified.
+        def create_consumer_config
+          config_file = 'deimos.rb'
+          config_file_path = "#{initializer_path}/#{config_file}"
+          config_file_path = config_path if config_path.present?
+
+          if File.exist?(config_file_path)
+            config_template = File.expand_path(find_in_source_paths('config.rb'))
+            insert_into_file(config_file_path.to_s, CapturableERB.new(::File.binread(config_template)).result(binding))
+          else
+            template('config.rb', config_file_path.to_s)
+          end
+        end
+
+        # Generates schema classes
+        def create_deimos_schema_class
+          Deimos::Generators::SchemaClassGenerator.start
+        end
+
+        # @return [String]
+        def db_migrate_path
+          if defined?(Rails.application) && Rails.application
+            paths = Rails.application.config.paths['db/migrate']
+            paths.respond_to?(:to_ary) ? paths.to_ary.first : paths.to_a.first
+          else
+            'db/migrate'
+          end
+        end
+
+        # @return [String]
+        def migration_version
+          "[#{ActiveRecord::Migration.current_version}]"
+        rescue StandardError
+          ''
+        end
+
+        # @return [String]
+        def table_class
+          table_name.classify
+        end
+
+        # @return [String]
+        def schema
+          last_dot = self.full_schema.rindex('.')
+          self.full_schema[last_dot + 1..-1]
+        end
+
+        # @return [String]
+        def namespace
+          last_dot = self.full_schema.rindex('.')
+          self.full_schema[0...last_dot]
+        end
+
+        # @return [Deimos::SchemaBackends::Base]
+        def schema_base
+          @schema_base ||= Deimos.schema_backend_class.new(schema: schema, namespace: namespace)
+        end
+
+        # @return [Array<SchemaField>]
+        def fields
+          schema_base.schema_fields
+        end
+
+        # @return [String]
+        def table_name
+          schema.tableize
+        end
+
+        # @return [String]
+        def model_name
+          table_name.underscore.singularize
+        end
+
+        # @return [String]
+        def consumer_name
+          "#{schema.classify}Consumer"
+        end
```

Comment:
A lot of this is copied from the migration template. There should be a way to just call the migration generator from this generator rather than having the same code twice.

Reply:
I considered extending from the migration template but decided it wasn't the right approach. Will find a better way!